scripts/save_avro.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""
+Script to save data from JSON files to an Avro file
+
+Usage:
+    $ python save_avro.py ingestion_document_id codec output_directory
+
+To use snappy support you need to install snappy with your package manager,
+    and pip install python-snappy
+
+Homebrew package is "snappy", in Ubuntu you need libsnappy1 and libsnappy-dev
+
+"""
+import os
+import sys
+import argparse
+import avro.schema
+from avro.datafile import DataFileWriter
+from avro.io import DatumWriter
+
+from amara.thirdparty import json
+from dplaingestion.couch import Couch
+from dplaingestion.selector import getprop
+
+avro_schema = """
+{
+    "namespace": "ingestion1.avro",
+    "type": "record",
+    "name": "record",
+    "fields": [
+        { "name": "id",         "type": "string" },
+        { "name": "document",   "type": "string" }
+    ]
+}"""
+
+
+def define_arguments():
+    """Defines command line arguments for the current script"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "ingestion_document_id",
+        help="The ID of the ingestion document"
+    )
+    parser.add_argument(
+        "codec",
+        help="One of null, deflate, snappy (if installed)."
+    )
+    parser.add_argument(
+        "output_file",
+        help="The folder where the Avro records will save"
+    )
+    return parser
+
+
+def main(argv):
+    parser = define_arguments()
+    args = parser.parse_args(argv[1:])
+
+    schema = avro.schema.parse(avro_schema)
+    writer = DataFileWriter(
+        open(args.output_file, "w"),
```

Comment:
Refactoring would mean putting this below the CouchDB lookup (which might be something to do anyway).